src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,29 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+void
+cockpit_cpu_temperature (CockpitSamples *samples)
+{
+  gchar *contents = NULL;
+  GError *error = NULL;
+  gsize len;
+  guint32 temperature;
+
+  // file does not exist in virtual machines
+  if (g_file_test ("/sys/devices/virtual/thermal/thermal_zone0/temp", G_FILE_TEST_EXISTS))
```

Comment:
This is also a bit of a bad pattern — rather just try to open the file.  If you get a `ENOENT` (see `g_error_matches()`) then ignore it, otherwise report it.

Reply:
And: how do we know for sure that zone0 is definitely the CPU?

Reply:
```sh
~ ❱ tail /sys/devices/virtual/thermal/thermal_zone*/type
==> /sys/devices/virtual/thermal/thermal_zone0/type <==
acpitz

==> /sys/devices/virtual/thermal/thermal_zone1/type <==
INT3400 Thermal

==> /sys/devices/virtual/thermal/thermal_zone2/type <==
SEN1

==> /sys/devices/virtual/thermal/thermal_zone3/type <==
pch_cannonlake

==> /sys/devices/virtual/thermal/thermal_zone4/type <==
B0D4

==> /sys/devices/virtual/thermal/thermal_zone5/type <==
iwlwifi_1

==> /sys/devices/virtual/thermal/thermal_zone6/type <==
x86_pkg_temp
```

0 is `acpitz` ("ACPI thermal zone")

At least here, my CPU "package" (`x86_pkg_temp`: CPU, iGPU, etc.) is zone6.

![image](https://user-images.githubusercontent.com/10246/160802523-13ca7019-b522-400b-948a-02dfc66c7014.png)


Temperatures differ quite a bit too:

```
~ ❱ tail /sys/devices/virtual/thermal/thermal_zone*/temp
==> /sys/devices/virtual/thermal/thermal_zone0/temp <==
52000

==> /sys/devices/virtual/thermal/thermal_zone1/temp <==
20000

==> /sys/devices/virtual/thermal/thermal_zone2/temp <==
40050

==> /sys/devices/virtual/thermal/thermal_zone3/temp <==
51000

==> /sys/devices/virtual/thermal/thermal_zone4/temp <==
52050

==> /sys/devices/virtual/thermal/thermal_zone5/temp <==
42000

==> /sys/devices/virtual/thermal/thermal_zone6/temp <==
51000
```

However, right now, zone 0 is 52 and zone 6 is 51.

(I also wonder what Windows and macOS show for temperatures.)